src/main/java/io/neonbee/internal/helper/CollectionHelper.java

```diff
@@ -138,6 +139,21 @@ public static <T> T copyOf(T object) {
         }
     }
 
+    /**
+     * Converts a given map {@link Map} to a case-insensitive treemap {@link TreeMap} by creating a copy of all mutable
+     * keys and values and adding them to a new case-insensitive treemap {@link TreeMap}.
+     *
+     * @param <K> the key type
+     * @param <V> the value type
+     * @param map the map to copy
+     * @return a new case-insensitive treemap as mutable deep copy of the given map
+     */
+    public static <K extends String, V> Map<K, V> mapToCaseInsensitiveTreeMap(Map<K, V> map) {
+        return Optional.ofNullable(map).map(Map::entrySet).map(Set::stream).orElseGet(Stream::empty)
+                .collect(Collectors.toMap(entry -> copyOf(entry.getKey()), entry -> copyOf(entry.getValue()),
```

Comment:
With null values we would currently run into an NPE because Collectors.toMap calls merge and the TreeMap does not override the default method of java.util.Map#merge.
For example see io.neonbee.internal.helper.CollectionHelper.NullLiberalMergingHashMap

Reply:
Unfortunately, the test for this is missing. Could you add this test to your PR?
```
package io.neonbee.internal.helper;

import static com.google.common.truth.Truth.assertThat;

import java.util.HashMap;
import java.util.Map;
import java.util.TreeMap;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class CollectionHelperTest {

    @Test
    @DisplayName("Copy map with mutableCopyOf")
    void testMutableCopyOf() {
        HashMap<String, Object> map = new HashMap<>();
        map.put("k", null);
        map.put("key", "value");
        map.put("k1", 1);
        Map<String, Object> resultMap = CollectionHelper.mutableCopyOf(map);
        assertThat(resultMap).isInstanceOf(HashMap.class);
        assertThat(resultMap).containsExactlyEntriesIn(map);

    }

    @Test
    @DisplayName("Copy map with mapToCaseInsensitiveTreeMap")
    void testMapToCaseInsensitiveTreeMap() {
        HashMap<String, Object> map = new HashMap<>();
        map.put("k", null);
        map.put("key", "value");
        map.put("k1", 1);
        Map<String, Object> resultMap = CollectionHelper.mapToCaseInsensitiveTreeMap(map);
        assertThat(resultMap).isInstanceOf(TreeMap.class);
        assertThat(resultMap).containsExactlyEntriesIn(map);
    }
}
```

Reply:
Thank you for review, I've made requested changes.